php-frontend/src/test/java/org/sonar/php/PHPAnalyzerTest.java

```diff
@@ -65,29 +63,26 @@ class PHPAnalyzerTest {
   private final SensorContextTester sensorContext = SensorContextTester.create(Paths.get("."));
 
   @RegisterExtension
-  private final LogTesterJUnit5 logTester = new LogTesterJUnit5();
-  private final TemporaryFolder tmpFolder = new TemporaryFolder();
+  public final LogTesterJUnit5 logTester = new LogTesterJUnit5();
 
-  @BeforeEach
-  void before() throws IOException {
-    tmpFolder.create();
-  }
+  @TempDir
+  public File tempFolder;
 
   @Test
-  void parsingFailureShouldRaiseAnException() throws IOException {
+  void parsingFailureShouldRaiseAnException() {
     PHPCheck check = new DummyCheck();
     PHPAnalyzer analyzer = createAnalyzer(check);
-    InputFile file = FileTestUtils.getInputFile(tmpFolder.newFile(), "<?php if(condition): ?>");
+    InputFile file = FileTestUtils.getInputFile(new File(tempFolder, "file"), "<?php if(condition): ?>");
     Assertions.assertThrows(RecognitionException.class, () -> {
```

Comment:
It can be replaced by `Assertions.catchThrowable()` or `assertThatExceptionOfType().isThrownBy() `.
Similar in `SeparatedListImplTest`